Malware Archive (#110)
CybercentreCanada/assemblyline4_docs: docs/installation/configuration/malware_archive.md

# Malware Archive section

> Configuring the Malware Archive functionnality is done by modifying the Database section (`datastore:`) and the file storage section (`filestore:`). Refer to the [Malware Archive](../overview/architecture/#keeping-files-forever-malware-archive) to get an understanding of how it works.

Reviewer: ```suggestion
Configuring the Malware Archive functionality is done by modifying the data storage section (`datastore:`) and the file storage section (`filestore:`). Refer to the [Malware Archive](../overview/architecture/#keeping-files-forever-malware-archive) to get an understanding of how this feature works.
```

> Enable the archive option in the configuration to show the "Archive" link in the left navbar, active the Archiver core component and create the archive indices in the databse. The indices list defines which indices will have an archive counterpart can be omitted since the defaults.

Reviewer: ```suggestion
Enable the `archive:` option in the configuration to show the "Archive" link in the left navbar, activate the Archiver core component and create the archive indices in the database. The `indices:` list defines which indices will have an archive counterpart. This key is not required given its default values of `file`, `submission`, and `result`.
```

???+ example "Datastore section configuration example"
    ```yaml
    ...
    datastore:
      # Datastore Archive feature configuration
      archive:
        # Are we enabling Achiving features across indices?
        enabled: true

        # List of indices the ILM Applies to
        indices:
        - file
        - submission
        - result

    filestore:
      # List of filestores used for malware archive
      archive:
      - s3://al_storage_key:Ch@ngeTh!sPa33w0rd@localhost:9000?s3_bucket=al-archive&use_ssl=False
    ...
    ```

!!! tip

>     Refer to the [changing the configuration file](../config_file/#changing-the-configuration-file) documentation for more detail on where and how to change the configuration of the system.

Reviewer: ```suggestion
    Refer to the [changing the configuration file](../config_file/#changing-the-configuration-file) documentation for more details on where and how to change the configuration of the system.
```
